src/Symfony/Component/Console/Formatter/WrappableOutputFormatterInterface.php

```diff
@@ -15,11 +15,23 @@
  * Formatter interface for console output that supports word wrapping.
  *
  * @author Roland Franssen <franssen.roland@gmail.com>
+ * @author Krisztián Ferenczi <ferenczi.krisztian@gmail.com>
  */
 interface WrappableOutputFormatterInterface extends OutputFormatterInterface
 {
     /**
      * Formats a message according to the given styles, wrapping at `$width` (0 means no wrapping).
      */
     public function formatAndWrap(string $message, int $width);
+
+    /**
+     * Separate word wrapping method.
+     *
+     * @param string   $message
+     * @param int      $width
+     * @param int|null $cutOption
+     *
+     * @return string
+     */
+    public function wordwrap(string $message, int $width, int $cutOption = null): string;
```

Comment:
This is new feature isnt it? To wrap a raw string with tags (unformatted still) ...

Do we really need it? As such it's a BC break, and it's the exact same reason we added a new `WrappableOutputFormatterInterface` in 4.2.

New features target master and thus would imply another interface. I think we should avoid that and solve the issue in `formatAndWrap()`.

Which takes me to my 2nd point; currently we parse the tags twice, my main motivation for "formatAndWrap" was to be able to handle it at once. I think we should patch it instead, relying on a helper if really needed for clarity.

Reply:
I was thinking about it also. I saw 3 ways:
1. I add the CUT option parameter to the `formatAndWrap()` --> this causes BC break
2. I create a new wrap function that only can wrap --> it doesn't cause BC break
3. I leave the CUT option parameter --> it doesn't cause BC break, but it causes inflexible wrapping function

Maybe there is a 4. option: I remove `wordwrap()` (maybe the `formatAndWrap()` too), and if somebody want to wrap then call like here:

```php
// Without wrapping
$output->getFormatter()->format($text);

// With wrapping
$output->getFormatter()->format(
    PrettyWordWrappingHelper::wordwrap($text, 120)
);
```

It would be clearer, but it supersede your `WrappableOutputFormatterInterface` because wrapping and formatting are really and fully separated (and helps to programmers changing it a custom wrapper if they want) I would prefer this.

> this PR will parse the tags twice

Yes, it is true. This is a side effect, which I think is necessary. Formatting (styling?) and wrapping can enough complex separately, that is why I think we should handle it separately. It causes more maintenance and extendable code. And this will be used in command line, the performance isn't so important. What do you think?

Reply:
option 5) Your  "With wrapping" example to me equals `formatter->formatAndWrap($text, 120)` already. That's API in 4.2 we can fix, maybe even with an `@internal` WordwrapHelper, or public for 4.3/master.

option 6) deprecate `WrappableOutputFormatterInterface` in 4.3 and add the "best" API directly to OutputFormatterInterface using #28902 (which wasnt possible back then)

In general i think the wordwrap helper should be an implementation detail of the output formatter. But we need some consensus first to move forward. Given a lot of work is done here already, perhaps we should finish it :) The bug is real.

> this PR will parse the tags twice
>> Yes, it is true. This is a side effect, which I think is necessary. 

You know best :) so far i thought it seemed reasonably possible ... but it's also still flawed currently :sweat_smile: I need to look closer at this new approach to see how it differs.

Reply:
... I'm thinking on it ...